pkg/platform/runtime/installer.go

```diff
@@ -195,9 +195,9 @@ func (installer *Installer) InstallArtifacts(runtimeAssembler Assembler) (envGet
 		}
 	}
 
-	downloadArtfs, unpackArchives := runtimeAssembler.ArtifactsToDownloadAndUnpack()
+	downloadArtfs := runtimeAssembler.ArtifactsToDownload()
 
-	if len(downloadArtfs) == 0 && len(unpackArchives) == 0 {
+	if len(downloadArtfs) == 0 {
 		// Already installed, no need to download or install
 		logging.Debug("Nothing to download")
 		return runtimeAssembler, false, nil
```

Comment:
Note that this block cancels *all* installation steps ie., also the `PreInstall()` where potentially artifacts are being removed.

So, if there is nothing to be downloaded but something is to be removed, we want to continue here...